File: sentry-java/r8-throwable-class-merging/app/src/main/java/com/example/myapp/MainActivity.java
package com.example.myapp;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public final class MainActivity extends Activity {

    private static final String TAG = "MainActivity";

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final State loading = State.Loading.INSTANCE;
        final State error = State.Error.INSTANCE;
        final State success = new State.Success("Hello World");

        try {
            loading.goNuts();
        } catch (Exception e) {
            Log.d(TAG, "loading", e);
        }
        try {
            error.goNuts();
        } catch (Exception e) {
            Log.d(TAG, "error", e);
        }
        try {
            success.goNuts();
        } catch (Exception e) {
            Log.d(TAG, "success", e);
        }
        final TextView output = new TextView(this);
        output.setPadding(72, 72, 72, 72);
        output.setText(
                "loading: " + loading.getClass() + "\n" +
                "error: " + error.getClass() + "\n" +
                "success: " + success.getClass());
        setContentView(output);
    }
}

File: sentry-java/5980/app/src/main/java/io/sentry/repro/issue5980/MainActivity.kt
package io.sentry.repro.issue5980

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.widget.ScrollView
import android.widget.TextView
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

const val TAG = "Repro5980"

class MainActivity : Activity() {

  private lateinit var output: TextView

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    output = TextView(this).apply { setPadding(24, 24, 24, 24) }
    setContentView(ScrollView(this).apply { addView(output) })

    // adb shell am start -n io.sentry.repro.issue5980/.MainActivity \
    //   --ei writers 8 --ei readers 8 --ei ops 60
    val workload =
      StartupWorkload(
        context = applicationContext,
        writers = intent.getIntExtra("writers", 8),
        readers = intent.getIntExtra("readers", 8),
        opsPerWorker = intent.getIntExtra("ops", 60),
        warmUp = intent.getBooleanExtra("warmup", false),
      )

    CoroutineScope(Dispatchers.Default).launch {
      val report = workload.run { line -> log(line) }
      withContext(Dispatchers.Main) { output.text = report }
    }
  }

  private suspend fun log(line: String) {
    Log.i(TAG, line)
    withContext(Dispatchers.Main) { output.append(line + "\n") }
  }
}
